### PR TITLE
Update: Add autofix for `sort-vars`

### PR DESCRIPTION
**Pull request by thofmann**
**What is the purpose of this pull request?**

- [x] Add autofixing to a rule

**What changes did you make?**

The `sort-vars` rule can now automatically fix unsorted variable declarations when each variable is either unassigned or assigned to a literal value. I added the implementation for this fixer, test cases to validate that the fixer works as expected, and test cases to validate that it does not fix when a variable is assigned to a non-literal value.

In short, the fixer will convert `var c, b, a` to `var a, b, c` and `var b = 3, a = true` to `var a = true, b = 3` but **will not** fix `var b = f(), a = g()` or `var b = 5, a = b`.

**Is there anything you'd like reviewers to focus on?**

All of it, please :confused:

### CODE REVIEW

File: tests/lib/rules/sort-vars.js [hunk 1]
```diff
@@ -113,34 +113,109 @@ ruleTester.run("sort-vars", rule, {
...
+        },
+        {
+            code: "var b, c, a, d",
+            output: "var a, b, c, d",
+            errors: [expectedError]
+        },
+        {
+            code: "var c, d, a, b",
+            output: "var a, b, c, d",
+            errors: 2
```
Comment (Aladdin-ADD) [inline tests/lib/rules/sort-vars.js]: is this intentional?

File: tests/lib/rules/sort-vars.js [hunk 2]
```diff
@@ -113,34 +113,109 @@ ruleTester.run("sort-vars", rule, {
...
+        },
+        {
+            code: "var b=10, a=f();",
+            output: null,
+            errors: [expectedError]
+        },
+        {
+            code: "var b=10, a=b;",
+            output: null,
+            errors: [expectedError]
```
Comment (Aladdin-ADD) [inline tests/lib/rules/sort-vars.js]: could you please add some more tests:
e.g.
```js
var b = 0, a = `${b}`;
var b = 0, a = `${foo()}`
var b = 0, c = b, a;
```

File: lib/rules/sort-vars.js [hunk 3]
```diff
@@ -27,29 +27,68 @@ module.exports = {
...
-                        currenVariableName = decl.id.name;
+                    const lastVariableName = getSortableName(memo),
+                        currentVariableName = getSortableName(decl);
 
-                    if (ignoreCase) {
-                        lastVariableName = lastVariableName.toLowerCase();
-                        currenVariableName = currenVariableName.toLowerCase();
-                    }
+                    if (currentVariableName < lastVariableName) {
+                        if (unfixable || fixed) {
```
Comment (Aladdin-ADD) [inline lib/rules/sort-vars.js]: Nitpick: maybe it make more sense to place this check to fixer function. (returning null means do not fix.)

File: tests/lib/rules/sort-vars.js (hunk 1 above)
Comment (thofmann) [reply tests/lib/rules/sort-vars.js]: Yes, this test case was already included; I only expanded it and added the expected fixed `output`.

It is expected to have two errors per the documentation:

> Alphabetical list is maintained starting from the first variable and excluding any that are considered problems. So the following code will produce two problems:
> `var c, d, a, b;`
> But this one, will only produce one:
> `var c, d, a, e;`

File: lib/rules/sort-vars.js (hunk 3 above)
Comment (thofmann) [reply lib/rules/sort-vars.js]: Thanks! I did not know about returning null in a fixer. I will change this.

File: tests/lib/rules/sort-vars.js (hunk 1 above)
Comment (Aladdin-ADD) [reply tests/lib/rules/sort-vars.js]: Do you mean this: `errors: [expectedError, expectedError]`

Comment (thofmann) [reply tests/lib/rules/sort-vars.js]: I'm a little confused. Are you suggesting that the rule's functionality be changed to only report one error?

Comment (Aladdin-ADD) [reply tests/lib/rules/sort-vars.js]: I guess reporting one error can simplify the code: 
`check is sorted => check is safe to fix => do fix`. 
maybe just my personal preference, I don't feel strong.

**Comment by platinumazure** [comment]
I would love to see the reporting logic be simplified to reporting one error per VariableDeclaration-- but eventually, and not in this PR. :smile:

**Comment by thofmann** [comment]
Agreed @platinumazure and @Aladdin-ADD - I would prefer simplified logic that reports only one error, but I did not want to change that functionality in this PR. :smiley:

**Comment by ilyavolodin** [comment]
@eslint/eslint-team Does anyone else wants to support this enhancement?

**Comment by thofmann** [comment]
If this does get merged, I would gladly submit a second PR to simplify the logic a bit and report only one error.

**Comment by platinumazure** [comment]
@eslint/eslint-team This just needs a champion and then we can accept. Otherwise, we should probably close this soon (though I really hope we don't).

**Comment by not-an-aardvark** [comment]
Thanks for contributing!
